vscode/src/services/HistoryChat.ts

```diff
@@ -0,0 +1,145 @@
+                    command: { command: 'cody.chat.panel.restore', args: [id, chatTitle] },
+                })
+            } else if (timeDiff < 48 * 60 * 60 * 1000) {
+                yesterdayChats.push({
+                    id,
+                    title: chatTitle,
+                    icon: 'comment-discussion',
+                    command: { command: 'cody.chat.panel.restore', args: [id, chatTitle] },
+                })
+            } else if (timeDiff < 7 * 24 * 60 * 60 * 1000) {
+                NDaysChats.push({
+                    id,
+                    title: chatTitle,
+                    icon: 'comment-discussion',
+                    command: { command: 'cody.chat.panel.restore', args: [id, chatTitle] },
+                })
+            } else if (timeDiff < 14 * 24 * 60 * 60 * 1000) {
+                lastWeekChats.push({
+                    id,
+                    title: chatTitle,
+                    icon: 'comment-discussion',
+                    command: { command: 'cody.chat.panel.restore', args: [id, chatTitle] },
+                })
+            } else if (timeDiff < 30 * 24 * 60 * 60 * 1000) {
+                NWeeksChats.push({
+                    id,
+                    title: chatTitle,
+                    icon: 'comment-discussion',
+                    command: { command: 'cody.chat.panel.restore', args: [id, chatTitle] },
+                })
+            } else if (timeDiff < 60 * 24 * 60 * 60 * 1000) {
+                lastMonthChats.push({
+                    id,
+                    title: chatTitle,
+                    icon: 'comment-discussion',
+                    command: { command: 'cody.chat.panel.restore', args: [id, chatTitle] },
+                })
+            } else {
+                NMonthsChats.push({
+                    id,
+                    title: chatTitle,
+                    icon: 'comment-discussion',
+                    command: { command: 'cody.chat.panel.restore', args: [id, chatTitle] },
+                })
+            }
+        }
+    })
+
+    return {
+        Today: todayChats.reverse(),
+        Yesterday: yesterdayChats.reverse(),
+        'Last Week': lastWeekChats.reverse(),
+        'Last month': lastMonthChats.reverse(),
+        'N days ago': NDaysChats.reverse(),
+        'N weeks ago': NWeeksChats.reverse(),
+        'N months ago': NMonthsChats.reverse(),
+    }
+}
+
+export async function displayHistoryQuickPick(): Promise<void> {
+    const groupedChats = groupCodyChats()
+    if (!groupedChats) {
+        return
+    }
+
+    const quickPickItems: HistoryItems[] = []
+
+    const addGroupSeparator = (groupName: string): void => {
+        quickPickItems.push({
+            label: groupName,
+            onSelect: async () => {},
+            kind: vscode.QuickPickItemKind.Separator,
+        })
+    }
+
+    for (const [groupName, chats] of Object.entries(groupedChats)) {
+        if (chats.length > 0) {
+            addGroupSeparator(groupName)
```

Comment:
Does this mean we start with a separator?
```suggestion
        if (chats.length > 0) {
            if (quickPickItems.length) {
              addGroupSeparator(groupName)
            }
```

or do it in addGroupSeparator, or use a `flatMap`, `join` to remove the blanks and add the separators.

Reply:
Yes, we start with a separator. `Today` would be the first separator. If we add that condition to check for the `quickPick` items length, the separator `today` won't be there so I think it is better to keep it this way.